src/premiumvideodevice/PremiumVideoStreamHandler.ts

```diff
@@ -0,0 +1,211 @@
+  private videoInput: HTMLVideoElement = document.createElement('video') as HTMLVideoElement;
+  private canvasInput: HTMLCanvasElement = document.createElement('canvas');
+  private inputCtx = this.canvasInput.getContext('2d');
+  private inputVideoStream: MediaStream | null = null;
+
+  // Outputs
+  private canvasOutput: HTMLCanvasElementWithCaptureStream = document.createElement(
+    'canvas'
+  ) as HTMLCanvasElementWithCaptureStream;
+  private outputCtx = this.canvasOutput.getContext('2d');
+  outputMediaStream: MediaStream = new MediaStream();
+
+  // Constructor just sets a logger for the object
+  constructor(private logger: Logger, private videoEffectDriver: PremiumVideoEffectDriver) {}
+
+  /**
+   * If existing, return the output media stream. If not existing, create and return
+   * a media stream off of our output canvas
+   * @returns MediaStream
+   */
+  getActiveOutputMediaStream(): MediaStream {
+    if (this.isOutputMediaStreamActive()) {
+      return this.outputMediaStream;
+    }
+    this.outputMediaStream = this.canvasOutput.captureStream(this.framerate);
+    this.cloneInputAudioTracksToOutput();
+    return this.outputMediaStream;
+  }
+
+  /**
+   * Configure an inputMediaStream so that it is placed onto a HTML video element and
+   * configure our transformation function (apply) to process as our stream continues
+   * loading new data
+   */
+  async setInputMediaStream(inputMediaStream: MediaStream | null): Promise<void> {
+    if (!inputMediaStream) {
+      this.stop();
+      return;
+    }
+
+    if (inputMediaStream.getVideoTracks().length === 0) {
+      this.logger.error('No video tracks in input media stream, ignoring');
+      return;
+    }
+
+    this.inputVideoStream = inputMediaStream;
+    const settings = this.inputVideoStream.getVideoTracks()[0].getSettings();
+    this.canvasOutput.width = settings.width;
+    this.canvasOutput.height = settings.height;
+    this.videoInput.addEventListener('loadedmetadata', this.process); // this will be the trigger function
+    this.videoInput.srcObject = this.inputVideoStream;
+    // avoid iOS safari full screen video -- not sure what this does
+    this.videoInput.setAttribute('playsinline', 'true');
+
+    this.videoInput.load();
+    try {
+      await this.videoInput.play();
+    } catch {
+      this.logger.warn('Video element play() overriden by another load()');
+    }
+
+    this.cloneInputAudioTracksToOutput();
+  }
+
+  /**
+   * Transform a singular frame from the input stream to apply desired
+   * video effects. Then place the output image data onto our output canvas
+   * configured with an output stream
+   */
+  process = async (_event: Event): Promise<void> => {
+    if (!this.inputVideoStream) {
+      return;
+    }
+    const processVideoStart = performance.now();
+
+    // Draw the videoInput onto our input canvas
+    if (this.videoInput.videoWidth) {
+      // Match input canvas to same dimensions as our input stream
+      if (
```

Comment:
we heck the conditions and if it doesn't match we set them match. I would just set these values directly. i.e. kept the line 105 and 106 only

Reply:
Switched